lib/health_checks/sidekiq.rb

```diff
@@ -15,18 +17,24 @@ def sidekiq(config)
   class LivenessServer
     LIVENESS_PORT = 8080
 
-    def start
+    def start(mongo_databases, redis_configs)
       Sidekiq::Logging.logger.info "Starting liveness server on #{LIVENESS_PORT}"
+
       Thread.start do
+        checks = mongo_databases.map { |db| Custom::MongoidCheck.new(db) }
+        checks += redis_configs.map{ |config| Custom::RedisCheck.new(config) }
+        checks << Custom::MemoryCheck.new
         server = TCPServer.new(LIVENESS_PORT)
         loop do
           Thread.start(server.accept) do |socket|
             begin
-              HealthChecks.memory
-              check_redis
-              check_mongo
+              threads = checks.map do |check|
+                Thread.new { check.run }
+              end
+              threads.each(&:join)
               respond_success(socket, 'Live!')
             rescue => e
```

Comment:
You can't capture an exception in one thread in another thread

Reply:
looks like Thread#join makes them reraise the exception. From the ruby docs it says that I have to use abort_on_exception=true to reraise the exception, but one guy from https://stackoverflow.com/questions/9095316/handling-exceptions-raised-in-a-ruby-thread says Thread#join does the job as well. And after some tests I can confirm that Thread#join reraise the exceptions. Yet, I have to use abort_on_exception=true, so the threads will stop execution after the first one has raised an exception.

Reply:
Good, good, reading works :)

Reply:
yeah, but after some tests I realized that with the option healthchecker=true the main Thread is killed in case of an exception. I think, therefore, the pod will also be killed on the spot even before the k8s livenessprobe handles the status 500 and kills subsequently the pod. Should it be an issue?

Reply:
I just saw the answer about dropping the concurrency stuff, so no issues for now :)